Assets/_BuildingBlocks/Credentials/Scripts/KeyVault.cs

```diff
@@ -153,83 +146,79 @@ private IEnumerator TryAllSupportedAuthorizations(Uri uri, string username, stri
         {
             foreach (var supportedAuthorization in supportedAuthorizationTypes.Keys)
             {
-                var potentialAuth = CreateStoredAuthorization(supportedAuthorization, uri, passwordOrKey, username); // pass usernameOrKey before username for constructor order
-                yield return TrySupportedAuthorization(potentialAuth, uri);
-                if (storedAuthorizations.ContainsKey(potentialAuth.Domain)) yield break; // if the Auth test was succesful, stop looking.
-            }
-        }
-
-        private IEnumerator TrySupportedAuthorization(StoredAuthorization.StoredAuthorization potentialAuth, Uri uri)
-        {
-            var config = Config.Default();
-            config = potentialAuth.AddToConfig(config);
-            var request = Uxios.DefaultInstance.Get<byte[]>(uri, config); //get as byte[] so Uxios doesn't try to interpret the response
+                // pass usernameOrKey before username for constructor order
+                var potentialAuth = CreateStoredAuthorization(
+                    supportedAuthorization, uri, passwordOrKey, username
+                ); 
 
-            bool isAuthorized = false;
+                var promisedAuthorization = TrySupportedAuthorization(potentialAuth, uri) as Promise<IResponse>;
 
-            request.Then(response => { isAuthorized = true; });
-            request.Catch(exception => { isAuthorized = CheckErrorAuthentication(exception); });
+                yield return Uxios.WaitForRequest(promisedAuthorization);
+                if (promisedAuthorization == null)
+                {
+                    throw new Exception("Expected promisedAuthorization to be casted to Promise<StoredAuthorization.StoredAuthorization>, but got null");
+                }
 
-            yield return Uxios.WaitForRequest(request);
+                // if the Auth test was successful, stop looking.
+                if (promisedAuthorization.CurState == PromiseState.Resolved) yield break;
```

Comment:
klopt dit wel met dat we nu een 400 response als geldige credentials beschouwen?

Reply:
ok, dan keur ik het goed, ik wilde even dubbel checken of PromiseState.Resolved ook voor 400 was